File: Python/Maths/GPA - Grade to Points calculation.py
grade = input("Enter the obtained grade: ")
if grade == "A+":
    points = 5.00
elif grade == "A":
    points = 4.00
elif grade == "A-":
    points = 3.50
elif grade == "B":
    points = 3.00
elif grade == "C":
    points = 2.00
elif grade == "D":
    points = 1.00
elif grade == "F":
    points = 0.00
else:
    points = -1


if points == -1:
    print("Invalid grade")
elif points == 0.00:
    print("\nResult Failed \nPoints:", round(points, 2))
else:
    print("\nResult Passed \nPoints:", round(points, 2))

File: Java/Maths/GPA - Grade to Points calculation.java
import java.util.Scanner;
class gpa_GradeToPoints {
    public static void main(String args[]){
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the obtained grade: ");
        String grade = input.nextLine();

        double points;
        if(grade.equals("A+")){
            points = 5.00;
        }else if(grade.equals("A")){
            points = 4.00;
        }else if(grade.equals("A-")){
            points = 3.50;
        }else if(grade.equals("B")){
            points = 3.00;
        }else if(grade.equals("C")){
            points = 2.00;
        }else if(grade.equals("D")){
            points = 1.00;
        }else if(grade.equals("F")){
            points = 0.00;
        }else{
            points = -1;
        }

        if(points == -1){
            System.out.println("Invalid grade");
        }else if(points == 0.00){
            System.out.printf("\nResult Failed \nPoints: %.2f", points);
        }else{
            System.out.printf("\nResult Passed \nPoints: %.2f", points);
        }
    }
}